src/main/java/core/basesyntax/service/impl/FruitTransactionCreationServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitTransactionCreationService;
+import java.util.List;
+
+public class FruitTransactionCreationServiceImpl implements FruitTransactionCreationService {
+    private static String operationValue = "b";
+    private static String fruitType = "";
+    private static int fruitQuantity = 0;
+    private final FruitDao dao;
+
+    public FruitTransactionCreationServiceImpl(FruitDao dao) {
+        this.dao = dao;
+    }
+
+    @Override
+    public void createTransactions(List<String[]> data) {
+        FruitTransaction transaction;
+        FruitTransaction.Operation operation = null;
+        String fruit;
+        int amount;
+        for (String[] parameters : data) {
+            transaction = new FruitTransaction();
+            operationValue = parameters[0];
+            fruitType = parameters[1];
+            fruitQuantity = Integer.parseInt(parameters[2]);
+            for (FruitTransaction.Operation op : FruitTransaction.Operation.values()) {
```

Comment:
in can be simplified because you have to compare actual and expected operations. no need to iterate through all operation values